### PR TITLE
RavenDB-16365 - creating better error message when using Properties inside Struct

### PR DESCRIPTION
**Pull request by omerlewitz**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-16365

### Additional description

_Please delete below the options that are not relevant_

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Extensions/MemberInfoExtensions.cs [hunk 1]
```diff
@@ -79,8 +79,7 @@ public static void SetValue<T>(this MemberInfo memberInfo, ref T entity, object
                 var propertyInfo = (PropertyInfo)memberInfo;
                 if (typeof(T).IsValueType)
                 {
-                    SetStructValue(propertyInfo.Name, propertyInfo.PropertyType, ref entity, value);
-                    return;
+                    throw new NotSupportedException("Properties are not allowed when using Struct. Use Fields instead.");
```
Comment (ayende) [inline src/Raven.Client/Extensions/MemberInfoExtensions.cs:82]: Isn't this going to have broader impact on the overall usage? Where else is this used?

Comment (omerlewitz) [reply src/Raven.Client/Extensions/MemberInfoExtensions.cs:82]: The only usage of `SetStructValue` is here.
I tested the functionality with `Properties` within a `Class` at it seems ok.
@karmeli87 what do you think?

Comment (karmeli87) [reply src/Raven.Client/Extensions/MemberInfoExtensions.cs:82]: We use the `SetValue` in `TrySetIdentityInternal` & `RefreshInternal`... 
But I'm not sure that those method would work with properties on structs.. please check this as well.

Comment (ayende) [reply src/Raven.Client/Extensions/MemberInfoExtensions.cs:82]: `TrySetIdentityInternal` works with `object`, so structs aren't relevant here.
`Refresh` _could_ work on structs, I guess, but likely very rare scenario, and mostly probably a wrong one

**Review by ayende — APPROVED** [review]

**Pull request merged**